changelogs/unreleased/gh-4726-execute-field-in-box.md

```diff
@@ -0,0 +1,3 @@
+## feature/box
```

Comment:
I wouldn't call it a feature. Neither a bugfix really, but a bugfix fits better. Also this needs to be marked as a breaking change. Because there is at least one customer, who expected `box.execute` to be available before `box.cfg`: https://github.com/tarantool/tarantool/issues/4231. If his code still exists like that, then it will break.

Reply:
I would ask @R-omk directly, but I guess that it was convenient for him to *capture* `box.execute` before `box.cfg()`, while the ability to *call* it before `box.cfg()` was irrelevant. I think we can keep ability to capture `box.execute` before `box.cfg()`, but raise an error instead of automatic box initialization.

I'm also curious, whether `local box_execute = box.execute` offers any benefits. Performance? Style consistency?

I saw considerable amount of old code, which doing something like `local gsub = string.gsub` at top of a module, but I doubt that it has any sense for LuaJIT. I saw [this](https://gitspartv.github.io/LuaJIT-Benchmarks/#test2) benchmark, but it counts difference in instructions outside of the loop. The real performance is the same (at least for this test). Curious fact: if I change `sin(3.14)` to `sin(i)`, the loops in both code variants are still equivalent.

Reply:
> I think we can keep ability to capture box.execute before box.cfg(), but raise an error instead of automatic box initialization.

BTW, it is not the only way to handle this case. We can also consider a yield till box initialization.

Reply:
> I think we can keep ability to capture box.execute before box.cfg(), but raise an error instead of automatic box initialization.

I agree with it, but it doesn't matter to me now.

Reply:
Thanks all for the review and comments! I still suggest to drop this method. This method doesn't something that can live without initialized `box`.

Reply:
> I wouldn't call it a feature. Neither a bugfix really, but a bugfix fits better. Also this needs to be marked as a breaking change. Because there is at least one customer, who expected `box.execute` to be available before `box.cfg`: #4231. If his code still exists like that, then it will break.

Thanks, fixed.

Reply:
I asked @Mons (as product manager). He said that `box.execute` should behave like anything else and so a user should get 'Please call box.cfg{} first' error at attempt to acquire this field before `box.cfg()`.

Reply:
Thanks for clarifying this issue!

Reply:
*(I'm fine with the result of the discussion, but I left it open, because it was opened by Vlad.)*

Reply:
Filed https://github.com/GitSparTV/LuaJIT-Benchmarks/issues/3.